htdocs/compta/facture/card.php

```diff
@@ -2048,6 +2048,47 @@
 					if ($ret > 0) {
 						$nextSituationInvoice->insertExtraFields();
 					}
+
+					// Hooks
+					$regs = array();
```

Comment:
this line until 2085 should be done inside the hook.
No reason to overload code with this when there is no hook...

Reply:
I used the same code as the other invoices (line 1907 for the hook and lines 1507 to 1562 for the creation of $srcobject)